src/backend/dx11/src/device.rs

```diff
@@ -182,9 +183,26 @@ impl Device {
             vertex_strides[buffer.binding as usize] = buffer.stride;
         }
 
+        // See [`shader::introspect_spirv_vertex_semantic_remapping`] for details of why this is needed.
+        let semantics: Vec<_> = attributes.iter().map(|attrib| {
```

Comment:
it looks like constructing the `semantics :Vec` is not necessary here. Can't we just derive the semantic for each attribute on the spot within the `.filter_map(|attrib| {}` body?